### PR TITLE
Auto hide menu bar preference

### PR DESCRIPTION
**Pull request by ajivoin**
## Contents
- Adds preferences menu to house this preference and future ones.
- Adds "Toggle Auto Hide Menu Bar" option to the preferences menu. It has a checkbox to display the value of this preference.
- Adds dependency on `electron-settings` to save user preferences. It's a very simple-to-use package, I recommend using it to manage user preferences. It stores in the OS default appData directory.

Please discuss and make suggestions!

Created because of #6

### CODE REVIEW

**Comment by chrisknepper** [comment]
Hey @ajivoin , thanks for doing this. I think it looks great, there's just a couple things I think we should change. I'll make code comments and if we get them squared away, I'm happy to merge!

File: package.json [hunk 1]
```diff
@@ -64,7 +65,7 @@
     "babel-plugin-transform-object-rest-spread": "^7.0.0-beta.3",
     "chai": "^4.1.0",
     "css-loader": "^0.28.7",
-    "electron": "2.0.2",
+    "electron": "^2.0.2",
```
Comment (chrisknepper) [inline package.json]: Can you undo this change? The `electorn-boilerplate` [recommends specifying an exact version here](https://github.com/szwacz/electron-boilerplate#upgrading-electron-version) and they point to the Electron docs.

File: src/menu/pref_menu.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import settings from "electron-settings";
+
+export const prefMenu = {
+  label: "Preferences",
```
Comment (chrisknepper) [inline src/menu/pref_menu.js]: Can you change this to `Settings`? (since macOS has a very specific meaning for the word "Preferences" and that item usually lives under the main menu in macOS, which could confuse macOS users)

File: src/background.js [hunk 3]
```diff
@@ -41,6 +43,7 @@ if (isSecondInstance) {
     if (env.name !== 'production') {
       menus.push(devMenuTemplate);
     }
+    menus.push(prefMenu);
```
Comment (chrisknepper) [inline src/background.js]: Instead of doing this here, could you push this inside an `if (!IS_MAC)` guard inside `base_menu_template.js`. As far as I know, hiding the menu bar has no effect on macOS because the menu bar on macOS is at the top of the screen, not the application.

As an aside, I'm also wondering if any Linux users know if this has an effect on Linux.

Comment (chrisknepper) [reply src/background.js]: Also, would you mind renaming this to keep the same naming structure of the menu files (`_template.js`)?

File: src/background.js [hunk 4]
```diff
@@ -61,12 +64,23 @@ if (isSecondInstance) {
   }
 
   app.on('ready', () => {
+    let autoHideMenuPref = false;
+    if (!settings.has("autoHideMenuPref")) {
```
Comment (chrisknepper) [inline src/background.js]: Can you also wrap these operations in an `if (!IS_MAC)`

File: src/menu/dev_menu_template.js [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import { app, BrowserWindow } from "electron";
+import settings from "electron-settings";
```
Comment (chrisknepper) [inline src/menu/dev_menu_template.js]: Can we get rid of this import?

File: src/helpers/window.js [hunk 6]
```diff
@@ -79,6 +79,6 @@ export default (name, options) => {
   win = new BrowserWindow(Object.assign({}, options, state));
 
   win.on("close", saveState);
-
+
```
Comment (chrisknepper) [inline src/helpers/window.js]: Can we get rid of this space?

File: src/menu/help_menu_template.js [hunk 7]
```diff
@@ -8,7 +8,7 @@ export const helpMenuTemplate = {
     {
         label: 'Learn More',
         click: () => shell.openExternal('https://github.com/chrisknepper/android-messages-desktop/')
-    }
+    },
```
Comment (chrisknepper) [inline src/menu/help_menu_template.js]: Can we get rid of this comma?

**Comment by ajivoin** [comment]
@chrisknepper I'll fix it up tomorrow! I can attempt to test it in Linux and let you know how it behaves

**Comment by ajivoin** [comment]
@chrisknepper I believe I addressed all of your comments, let me know if there is anything else I need to change

**Comment by chrisknepper** [comment]
@ajivoin Looks great! Did you get the chance to test in Linux? I'm installing Ubuntu right now just to take a quick peak at it, but I think we can merge shortly.

**Comment by chrisknepper** [comment]
Just tested on Ubuntu, thanks again for your great work!

**Comment by chrisknepper** [comment]
Will create a release either tonight or tomorrow.
